src/main/java/com/ccs/inventorymanagement/route/CreateClothingHandler.java

```diff
@@ -0,0 +1,94 @@
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class CreateClothingHandler implements HandlerFunction<ServerResponse> {
+
+    private ClothingService clothingService;
+
+    public CreateClothingHandler(ClothingService clothingService) {
+        this.clothingService = clothingService;
+    }
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(Request.class)
+                .flatMap(request -> clothingService.create(Clothing.builder()
+                        .name(request.getName())
+                        .condition(request.getCondition())
+                        .status(request.getStatus())
+                        .description(request.getDescription())
+                        .brand(request.getBrand())
+                        .color(request.getColor())
+                        .type(request.getType())
+                        .gender(request.getGender())
+                        .size(request.getSize())
+                        .build()))
+                .flatMap(clothing -> ServerResponse.ok()
+                        .body(BodyInserters.fromValue(Response.from(clothing))));
+    }
+
+    @Data
+    @Builder
+    @AllArgsConstructor
+    public static final class Request {
+        private String name;
+        private final Item.Condition condition;
+        private boolean present;
+        private final Instant created;//this should not change
+        private Instant updated;
+        private Item.Status status;
+        private String description;
+        private String brand;
+        private String color;
+        private Clothing.Type type;
+        private Clothing.Gender gender;
+        private Clothing.Size size;
+    }
+
+    @Data
+    @Builder
+    public static final class Response {
+        private UUID id;
+        private String name;
+        private final Item.Condition condition;
+        private boolean present;
+        private final Instant created;//this should not change
+        private Instant updated;
+        private Item.Status status;
+        private String description;
+        private String brand;
+        private String color;
+        private Clothing.Type apparelType;
+        private Clothing.Gender gender;
+        private Clothing.Size size;
+
+        public static Response from(Clothing clothing) {
+            return builder()
+                    .id(clothing.getId())
+                    .name(clothing.getName())
+                    .condition(clothing.getCondition())
+                    .status(clothing.getStatus())
+                    .description(clothing.getDescription())
+                    .brand(clothing.getBrand())
+                    .color(clothing.getColor())
+                    .apparelType(clothing.getType())
+                    .gender(clothing.getGender())
+                    .size(clothing.getSize())
+                    .build();
+        }
```

Comment:
Your response is missing some data, but if that is intentional, disregard.

Reply:
Fixed, please review.